### PR TITLE
enhancement(file source): Emit `StreamClosedError` to adhere to DroppedEvents spec

### PR DESCRIPTION
**Pull request by neuronull**
Closes:  #14149.
Part of: #13995.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]

File: src/sources/file.rs [hunk 1]
```diff
@@ -556,9 +556,19 @@ pub fn file_source(
...
         });
         tokio::spawn(async move {
-            out.send_event_stream(&mut messages)
+            match out
+                .send_event_stream(&mut messages)
                 .instrument(span.or_current())
                 .await
+            {
+                Ok(()) => {
+                    info!("Finished sending.");
```
Comment (spencergilbert) [inline src/sources/file.rs:565]: This feels like it would be noisy - should this be debug or trace instead?

Comment (neuronull) [reply src/sources/file.rs:565]: Good question- I modeled this after other cases also using `info()`.

These sources are doing that: `file_descriptors`,  `http_scrape`,  `apache_metrics` , `socket` , `syslog` . Though socket and syslog in a slightly different behavior.

Comment (spencergilbert) [reply src/sources/file.rs:565]: I think there were some suggestions that we wanted "regular logs to tell things are running" but I'm not sure where things are there. @jszwedko what do you think?

Comment (jszwedko) [reply src/sources/file.rs:565]: 👍 this seems like more of a `debug` level log to me.

Comment (neuronull) [reply src/sources/file.rs]: Addressed in https://github.com/vectordotdev/vector/pull/14155/commits/0350b29c539af48a71f2a0b7d91568018e32f07c

**Review by spencergilbert — APPROVED** [review]
Thanks for updating that log level! LGTM

**Pull request merged**